### PR TITLE
fix: Fix problem parsing mnt opt quotes

### PR DESCRIPTION
**Pull request by bfahr**
### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [X] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:
* Fixes the problem introduced in #3984 where there was no embedded comma in the quoted value then the option would not be saved.

### CODE REVIEW

File: insights/parsers/mount.py [hunk 1]
```diff
@@ -428,24 +428,31 @@ def _parse_mount_options(mount_options):
...
             opts[opt_name] = opt_value[1:-1]
+            in_quote = False
 
         # Else just a normal option or option=value
         else:
             if '=' in opt:
                 opt_name, opt_value = opt.split('=', 1)
-                opts[opt_name] = opt_value
+                # Check for quoted values and remove quotes if present
+                if not (opt_value.startswith('"') and opt_value.endswith('"')):
```
Comment (xiangce) [inline insights/parsers/mount.py:452]: @bfahr - I think you may have tried thinking of that, but I just want to ask: can the `opt_value.strip('"')` be used here?

File: insights/tests/parsers/test_mount.py [hunk 2]
```diff
@@ -61,6 +61,7 @@
 PROC_MOUNT_QUOTES = """
 /dev/mapper/rootvg-rootlv / ext4 rw,relatime,barrier=1,data=ordered 0 0
 tmpfs /var/lib/containers/storage/overlay-containers/ff7e79fc09c/userdata/shm tmpfs rw,nosuid,nodev,noexec,relatime,context="system_u:object_r:container_file_t:s0:c184,c371",size=64000k 0 0
+tmpfs /var/lib/containers/storage/overlay-containers/aa7e79fc09c/userdata/shm tmpfs rw,nosuid,nodev,noexec,relatime,context="system_u:object_r:container_file_t:s0",size=64000k 0 0
 """
```
Comment (JoySnow) [inline insights/tests/parsers/test_mount.py:65]: Hi @bfahr ,  
Seeing this adding  test case `...,context="system_u:object_r:container_file_t:s0",size=64000k`, I guess that maybe we can also be compatible for a mount option case like: 
`rw,nosuid,nodev,noexec,relatime,context="system_u:object_r:container_file_t:s0:c184,some_inside_quote_option,c371",size=64000k` 
What do you think ?

Comment (xiangce) [reply insights/tests/parsers/test_mount.py:65]: I think @JoySnow might mean: if we can add a test like `rw...,context="system_u:object_r:container_file_t:s0:c184,c371,c372",size=...`

Although the `context` was parsed correctly:
```python
assert device.mount_options.context == "system_u:object_r:container_file_t:s0:c184,c371,c372"
# it's True
```
However, the second `c371` was also parsed individually in the `else` part as a Boolean option
```python
assert  hasattr(device.mount_options, "c371") is True
assert device.mount_options.c371 is True
These are also `True`, but it should be not (right?)

```

File: insights/parsers/mount.py (hunk 1 above)
Comment (bfahr) [reply insights/parsers/mount.py:452]: @xiangce yes that would work, thanks for pointing that out.

File: insights/tests/parsers/test_mount.py (hunk 2 above)
Comment (bfahr) [reply insights/tests/parsers/test_mount.py:65]: Good catch @JoySnow!  I've added that test case and fixed the problem.

**Review by xiangce — APPROVED** [review]
lgtm

**Review by JoySnow — APPROVED** [review]

**Pull request merged**